addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -432,6 +432,7 @@ export const editorCommands = {
         }
     },
     removeFormat: editor => {
+        editor.historyPauseSteps();
```

Comment:
@Zynton  When the style is removed and we hit the undo button, the undo does not work as expected. The removal of the style is undone in an irregular manner. Therefore, to avoid non-uniform behavior, we pause the making of steps during the style removal phase.
 => [Video Reference](https://drive.google.com/file/d/1WHiIvzw7wiFjSYaM5sgKd2wahNoo-PDT/view)

Reply:
Was that video made on your branch before adding the `historyPauseSteps` or was it made in 15.0? I'd be interested in knowing what causes the issue because I see nothing in `removeFormat` that should be causing the creation of a history step in the middle of the process.

Reply:
@Zynton Yes the video is of my branch before adding the historyPauseSteps.

The `removeFormat` function is triggered by the data-call attribute with a click event in the `bindExecCommand` method. Clicking the `removeFormat` button executes this callback, which calls `execCommand` and `_applyCommand` to record `historySteps`. The `execCommand` is also invoked within `removeFormat`, triggering another `historyStep`.

Reply:
The `execCommand` invoked in `removeFormat` is not the same as the one invoked in the callback and it's not supposed to cause the creation of a history step. Notice that `removeFormat` doesn't call `editor.execCommand` but `editor.document.execCommand`, which is a [native javascript method](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand). So the problem must lie elsewhere. Since you say the event binding causes a step, you should also not have to call `historyStep` manually. In fact, I would expect your code to cause two steps instead of one.

Reply:
@Zynton I further looked into the problem and found that we are receiving a call on `_onInput` with `ev.inputType` as `formatRemove` once the `editor.document.execCommand` is executed, potentially creating a history step.

Reply:
Ah now that makes sense, thanks. In that case the `historyPauseSteps` makes sense but you explain this in a comment above it. Then unpausing the steps at the end should suffice to let `editor.execCommand` create the step after execution of this command. Can you confirm that is the case?

Reply:
@Zynton  Yes, we do not need to call `historyStep` after unpausing. Also, do I need to add a comment regarding why we used `historyPause/historyUnpause` in this scenario?

Reply:
Yes.
1. remove the unnecessary `historyStep`
2. add a comment for the adding of `historyPause/historyUnpause`

Reply:
@Zynton Removed the `historyStep` and added a comment.